src/ctl.c

```diff
@@ -2980,6 +2982,55 @@ arena_i_retain_grow_limit_ctl(tsd_t *tsd, const size_t *mib,
 	return ret;
 }
 
+static int
+arena_i_name_ctl(tsd_t *tsd, const size_t *mib, size_t miblen,
+    void *oldp, size_t *oldlenp, void *newp, size_t newlen) {
+	int ret;
+	unsigned arena_ind;
+	const char *name = NULL;
+
+	READ_XOR_WRITE();
+
+	malloc_mutex_lock(tsd_tsdn(tsd), &ctl_mtx);
+	MIB_UNSIGNED(arena_ind, 1);
+	if (arena_ind == MALLCTL_ARENAS_ALL || arena_ind >=
+	    ctl_arenas->narenas) {
+		malloc_printf("Invalid arena index %d. This function only takes in "
+		"indices no larger than %d.\n", arena_ind, ctl_arenas->narenas);
```

Comment:
We should not print anything, besides returning proper error codes. The user can decide to output if they'd like, however as a lower level lib let's not write anything unless have to (e.g. when terminating).

The other factor is, we should not say arena index greater than N is not valid, since it may change at any time, e.g. even before this msg makes it to the log.

Reply:
removed